### ISSUE TITLE
Training with 1-stage combined loss or in multi-stages?

### ISSUE THREAD

**Issue by bryanbocao** [Xinyu-Yi/TransPose#68]
Hi @Xinyu-Yi, thanks for sharing the amazing work. When training the whole model which consists of multiple stages each of which has its own loss function, do you (1) train the whole network in an end-to-end fashion with an overall loss function combining all stage losses, or (2) train each stage separately with its stage-wise loss?

Thanks!

**Comment by Xinyu-Yi** [comment]
Hi! Absolutely (2) train each stage separately with its stage-wise loss. No end-to-end fine-tuning in this work. But in my recent works I have tried to train the three stages together for fine-tuning and it improves the results a little.

**Comment by Xinyu-Yi** [comment]
No, I am not training the network using this `forward`. This is only used for debugging. I train each RNN separately using their own inputs and outputs and copy the weights here together.

**Comment by Xinyu-Yi** [comment]
All joint positions and orientations are expressed in the **root frame**. All accelerations and velocities are also expressed in the **root frame** without considering the non-inertial terms (which is corrected in my latest work PNP). Contacts do not need a reference frame. You can synthesize these values for each network, add Gaussian noise to the input, and train each network separately.

**Comment by bryanbocao** [comment]
@Xinyu-Yi Thanks for the detailed answers!

To keep the ```AMASS``` format consistent with ```DIP_IMU``` and ```TotalCapture```, I was thinking to simply add these lines of code
```
    # Save the format consistent with DIP_IMU and TotalCapture
    torch.save({'acc': out_vacc, 'ori': out_vrot, 'pose': out_pose, 'tran': out_tran}, os.path.join(paths.amass_dir, f'train.pt'))
    print('Synthetic AMASS dataset is saved at', paths.amass_dir)
```
after
https://github.com/Xinyu-Yi/TransPose/blob/4963e71ae33c3ea5ac24fcc053015804e9705ad1/preprocess.py#L85-L92
Do you think this is correct?

**Comment by bryanbocao** [comment]
> All joint positions and orientations are expressed in the **root frame**. All accelerations and velocities are also expressed in the **root frame** without considering the non-inertial terms (which is corrected in my latest work PNP). Contacts do not need a reference frame. You can synthesize these values for each network, add Gaussian noise to the input, and train each network separately.

Just a clarification question: by ***root frame*** you mean the coordinate system of the root joint (pelvis) device instead of local frame of each IMU device, right? @Xinyu-Yi

**Comment by Xinyu-Yi** [comment]
> > All joint positions and orientations are expressed in the **root frame**. All accelerations and velocities are also expressed in the **root frame** without considering the non-inertial terms (which is corrected in my latest work PNP). Contacts do not need a reference frame. You can synthesize these values for each network, add Gaussian noise to the input, and train each network separately.
> 
> Just a clarification question: by _**root frame**_ you mean the coordinate system of the root joint (pelvis) device instead of local frame of each IMU device, right? @Xinyu-Yi

Yes. The coordinate frame of the root (pelvis) joint. Not the frame of the imu device on the root joint. The frame is defined by SMPL (left-up-forward)
